src/metabase/models/dashboard.clj

```diff
@@ -69,48 +69,51 @@
+      (let [correct-card-ids     (->> (db/query {:select    [:dc.card_id]
                                                  :modifiers [:distinct]
-                                                 :from [[DashboardCard :dc]]
-                                                 :where [:and
-                                                         [:= :dc.dashboard_id dashboard-id]
-                                                         [:not= :dc.card_id nil]]})
+                                                 :from      [[DashboardCard :dc]]
+                                                 :where     [:and
+                                                             [:= :dc.dashboard_id dashboard-id]
+                                                             [:not= :dc.card_id nil]]})
                                       (map :card_id)
                                       set)
             stale-card-ids       (->> affected
-                                      (keep :card_id)
+                                      (keep :card-id)
                                       set)
             cards-to-add         (set/difference correct-card-ids stale-card-ids)
             card-id->dashcard-id (when (seq cards-to-add)
-                                   (db/select-field->id :card_id DashboardCard :dashboard_id dashboard-id,
+                                   (db/select-field->id :card_id DashboardCard :dashboard_id dashboard-id
                                                         :card_id [:in cards-to-add]))
             positions-for        (fn [pulse-id] (drop (pulse-card/next-position-for pulse-id)
                                                       (range)))
-            new-pulse-cards      (for [pulse-id                         pulse-ids,
-                                       [[card-id dashcard-id] position] (map vector card-id->dashcard-id
-                                                                                    (positions-for pulse-id))]
+            new-pulse-cards      (for [pulse-id                         pulse-ids
+                                       [[card-id dashcard-id] position] (map vector
+                                                                             card-id->dashcard-id
+                                                                             (positions-for pulse-id))]
                                    {:pulse_id          pulse-id
                                     :card_id           card-id
                                     :dashboard_card_id dashcard-id
                                     :position          position})]
         (db/transaction
-          (db/update-where! Pulse {:id [:in pulse-ids]}
+          (db/update-where! Pulse {:dashboard_id dashboard-id}
             :name (:name dashboard))
           (pulse-card/bulk-create! new-pulse-cards))))))
 
+(defn- post-update
+  [dashboard]
+  (update-dashboard-subscription-pulses! dashboard))
+
```

Comment:
always instructive to see little Cam improvements to code I wrote

Reply:
sorry, this wasn't really necessary, but I thought it made the `post-update` stuff a little clearer and wouldn't get hairy if we went and added more stuff to `post-update` in the future. Also it lets us test `update-dashboard-subscription-pulses!` by itself if we were so inclined

Reply:
I was referring to all of the minor edits above, not just pulling this function out specifically.